pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -560,6 +561,11 @@ func (m *managerImpl) evictPod(pod *v1.Pod, gracePeriodOverride int64, evictMsg
 			klog.Errorf("eviction manager: cannot get mirror pod from static pod %s, so cannot evict it", format.Pod(pod))
 			return false
 		}
+	} else if kubepod.IsMirrorPod(pod) {
+		if kubelettypes.IsCriticalPod(pod) {
+			klog.Errorf("eviction manager: cannot evict a critical mirror pod %s", format.Pod(pod))
+			return false
+		}
```

Comment:
As per this change, we will not be evicting a critical mirror pod as well. I am a little doubtful here. Let's discuss more before we proceed with this PR.

Reply:
All static pods are considered critical since 1.16. I think we should change `IsCriticalPod` to consider mirror pods as critical as well. Then, all we need here is to delete the whole `if IsStaticPod` statement and have just one check for the pod critical status and return false if the pod is critical.

Reply:
When kubepod.IsMirrorPod() is added to pod_update.go :
```
can't load package: import cycle not allowed
package k8s.io/kubernetes/pkg/kubelet/types
	imports k8s.io/kubernetes/pkg/kubelet/pod
	imports k8s.io/kubernetes/pkg/kubelet/types
```

Reply:
Please move `IsMirrorPod()` to pod_update.go and place it next to `IsStaticPod()`.

Reply:
> All static pods are considered critical since 1.16. I think we should change `IsCriticalPod` to consider mirror pods as critical as well. Then, all we need here is to delete the whole `if IsStaticPod` statement and have just one check for the pod critical status and return false if the pod is critical.

Hi, @bsalamat can this PR merged to 1.15.8 ?